File: Algorithms/src/3474. Lexicographically Smallest Generated String/Code/solution.go
func generateString(str1 string, str2 string) string {
	n := len(str1)
	m := len(str2)
	length := n + m - 1

	ans := make([]byte, length)
	fixed := make([]bool, length)

	for i := 0; i < length; i++ {
		ans[i] = '?'
	}

	// Apply all 'T' constraints
	for i := 0; i < n; i++ {
		if str1[i] == 'T' {
			for j := 0; j < m; j++ {
				pos := i + j

				if ans[pos] != '?' && ans[pos] != str2[j] {
					return ""
				}

				ans[pos] = str2[j]
				fixed[pos] = true
			}
		}
	}

	// Fill remaining positions with 'a'
	for i := 0; i < length; i++ {
		if ans[i] == '?' {
			ans[i] = 'a'
		}
	}

	// Process all 'F' constraints
	for i := 0; i < n; i++ {
		if str1[i] == 'F' {
			same := true

			for j := 0; j < m; j++ {
				if ans[i+j] != str2[j] {
					same = false
					break
				}
			}

			if !same {
				continue
			}

			changed := false

			// Try changing from right to left
			for j := m - 1; j >= 0; j-- {
				pos := i + j

				if fixed[pos] {
					continue
				}

				for c := byte('a'); c <= byte('z'); c++ {
					if c != ans[pos] && c != str2[j] {
						ans[pos] = c
						changed = true
						break
					}
				}

				if changed {
					break
				}
			}

			if !changed {
				return ""
			}
		}
	}

	return string(ans)
}

File: Algorithms/src/3474. Lexicographically Smallest Generated String/Code/solution.py
class Solution:
    def generateString(self, str1: str, str2: str) -> str:
        n = len(str1)
        m = len(str2)
        length = n + m - 1

        ans = ['?'] * length
        fixed = [False] * length

        # Apply all 'T' constraints
        for i in range(n):
            if str1[i] == 'T':
                for j in range(m):
                    pos = i + j

                    if ans[pos] != '?' and ans[pos] != str2[j]:
                        return ""

                    ans[pos] = str2[j]
                    fixed[pos] = True

        # Fill remaining positions with 'a'
        for i in range(length):
            if ans[i] == '?':
                ans[i] = 'a'

        # Process all 'F' constraints
        for i in range(n):
            if str1[i] == 'F':
                same = True

                for j in range(m):
                    if ans[i + j] != str2[j]:
                        same = False
                        break

                if not same:
                    continue

                changed = False

                # Try changing from right to left
                for j in range(m - 1, -1, -1):
                    pos = i + j

                    if fixed[pos]:
                        continue

                    for c in range(ord('a'), ord('z') + 1):
                        ch = chr(c)

                        if ch != ans[pos] and ch != str2[j]:
                            ans[pos] = ch
                            changed = True
                            break

                    if changed:
                        break

                if not changed:
                    return ""

        return ''.join(ans)